### PR TITLE
feat: add SSEConnection shared lib

### PR DESCRIPTION
**Pull request by jonathannorris**

### CODE REVIEW

**Comment by jonathannorris** [comment]
* **#838** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/838?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jonathannorris and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/shared/sse-connection/src/lib/SSEConnection.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                'StreamingConnection warning. Connection failed to establish.',
+            )
+        }
+        this.connection.onopen = () => {
+            this.logger.debug('StreamingConnection opened')
+        }
+    }
+
+    isConnected(): boolean {
+        return this.connection?.readyState === this.connection?.OPEN
```
Comment (JamieSinn) [inline lib/shared/sse-connection/src/lib/SSEConnection.ts:36]: this is so much cleaner and easier than having a threadsafe version :p

Comment (jonathannorris) [reply lib/shared/sse-connection/src/lib/SSEConnection.ts:36]: yea, this is just copying the JS SDK stuff into its own lib, still need to do some updates in a stacked PR for the NodeJS SDK.

File: lib/shared/sse-connection/src/lib/SSEConnection.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    reopen(): void {
+        if (!this.isConnected()) {
+            this.close()
+            this.openConnection()
```
Comment (JamieSinn) [inline lib/shared/sse-connection/src/lib/SSEConnection.ts:42]: Is a connection that's not connected required to be closed?
Do we care about connected sdks being told to close and then reconnect?

**Review by JamieSinn — APPROVED** [review]

File: lib/shared/sse-connection/src/lib/SSEConnection.ts (hunk 2 above)
Comment (jonathannorris) [reply lib/shared/sse-connection/src/lib/SSEConnection.ts:42]: yea, not sure yet, will figure that out when I try and integrate into NodeJS.

**Pull request merged**